### PR TITLE
Feature/chapter 9

### PR DESCRIPTION
**Pull request by gabrieladutra**

### CODE REVIEW

File: src/main/java/src/chapter9/Examples.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
+package src.chapter9;
+
+import java.lang.reflect.Array;
+import java.util.Arrays;
+
+public class Examples {
+  public static void print2D(int[][] mat) {
```
Comment (SerenityInAllThings) [inline src/main/java/src/chapter9/Examples.java:7]: Great printing tool! Congratulations!

File: src/main/java/src/chapter4/question1/Bird.java [hunk 2]
```diff
@@ -4,24 +4,15 @@
 package src.chapter4.question1;
```
Comment (SerenityInAllThings) [inline src/main/java/src/chapter4/question1/Bird.java]: Delete this file from this branch.

File: src/main/java/src/chapter4/question2/Bird.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package src.chapter4.question2;
```
Comment (SerenityInAllThings) [inline src/main/java/src/chapter4/question2/Bird.java]: Delete this file from this branch.

File: src/main/java/src/chapter4/question2/TestClass.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
+
```
Comment (SerenityInAllThings) [inline src/main/java/src/chapter4/question2/TestClass.java]: Delete this file from this branch.

File: src/main/java/src/chapter4/question3/Bird.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package src.chapter4.question3;
```
Comment (SerenityInAllThings) [inline src/main/java/src/chapter4/question3/Bird.java]: Delete this file from this branch.

File: src/main/java/src/chapter4/question3/TestClass.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
+package src.chapter4.question3;
```
Comment (SerenityInAllThings) [inline src/main/java/src/chapter4/question3/TestClass.java]: Delete this file from this branch.

File: src/main/java/src/chapter4/question4/Bird.java [hunk 7]
```diff
@@ -0,0 +1,4 @@
+package src.chapter4.question4;
```
Comment (SerenityInAllThings) [inline src/main/java/src/chapter4/question4/Bird.java]: Delete this file from this branch.

File: src/main/java/src/chapter5/question1/Question1.java [hunk 8]
```diff
@@ -0,0 +1,14 @@
+package src.chapter5.question1;
```
Comment (SerenityInAllThings) [inline src/main/java/src/chapter5/question1/Question1.java]: Delete this file from this branch.

File: src/main/java/src/chapter5/question2/Question2.java [hunk 9]
```diff
@@ -0,0 +1,15 @@
+package src.chapter5.question2;
```
Comment (SerenityInAllThings) [inline src/main/java/src/chapter5/question2/Question2.java]: Delete this file from this branch.

File: src/main/java/src/chapter5/question3/Question3.java [hunk 10]
```diff
@@ -0,0 +1,15 @@
+package src.chapter5.question3;
```
Comment (SerenityInAllThings) [inline src/main/java/src/chapter5/question3/Question3.java]: Delete this file from this branch.

File: src/main/java/src/chapter5/question4/Question4.java [hunk 11]
```diff
@@ -0,0 +1,19 @@
+package src.chapter5.question4;
```
Comment (SerenityInAllThings) [inline src/main/java/src/chapter5/question4/Question4.java]: Delete this file from this branch.

File: src/main/java/src/chapter9/question2/Question2.java [hunk 12]
```diff
@@ -0,0 +1,15 @@
+package src.chapter9.question2;
+
+import java.util.Arrays;
+
+public class Question2 {
+    public static int getArrayLength(int[] paramArray){
+        if(paramArray == null) throw new IllegalArgumentException("Array is null");
+        return paramArray.length;
+    }
+    public static void printArrayElement(int[] paramArray){
```
Comment (SerenityInAllThings) [inline src/main/java/src/chapter9/question2/Question2.java]: Change element to plural.

File: src/test/java/chapter4/Question2Test.java [hunk 13]
```diff
@@ -0,0 +1,39 @@
+package chapter4;
```
Comment (SerenityInAllThings) [inline src/test/java/chapter4/Question2Test.java]: Delete this file from this branch.

File: src/test/java/chapter5/Question1Test.java [hunk 14]
```diff
@@ -0,0 +1,19 @@
+package chapter5;
```
Comment (SerenityInAllThings) [inline src/test/java/chapter5/Question1Test.java]: Delete this file from this branch.

File: src/test/java/chapter5/Question2Test.java [hunk 15]
```diff
@@ -0,0 +1,24 @@
+package chapter5;
```
Comment (SerenityInAllThings) [inline src/test/java/chapter5/Question2Test.java]: Delete this file from this branch.

File: src/test/java/chapter5/Question3Test.java [hunk 16]
```diff
@@ -0,0 +1,26 @@
+package chapter5;
```
Comment (SerenityInAllThings) [inline src/test/java/chapter5/Question3Test.java]: Delete this file from this branch.

File: src/test/java/chapter5/Question4Test.java [hunk 17]
```diff
@@ -0,0 +1,21 @@
+package chapter5;
```
Comment (SerenityInAllThings) [inline src/test/java/chapter5/Question4Test.java]: Delete this file from this branch.

File: src/test/java/chapter7/Question2Test.java [hunk 18]
```diff
@@ -1,4 +1,4 @@
-package chapter7.question2;
+package chapter7;
```
Comment (SerenityInAllThings) [inline src/test/java/chapter7/Question2Test.java]: Delete this file from this branch.

**Review by SerenityInAllThings — CHANGES_REQUESTED** [review]
Some files modified/added in this branch are not related to chapter 9.
Question 4 is missing tests.
Questions 5-12 are missing.
Some adjustments required.

File: src/main/java/src/chapter4/question1/Bird.java (hunk 2 above)
Comment (gabrieladutra) [reply src/main/java/src/chapter4/question1/Bird.java]: File deleted.

File: src/main/java/src/chapter4/question2/Bird.java (hunk 3 above)
Comment (gabrieladutra) [reply src/main/java/src/chapter4/question2/Bird.java]: File deleted.

File: src/main/java/src/chapter4/question3/TestClass.java (hunk 6 above)
Comment (gabrieladutra) [reply src/main/java/src/chapter4/question3/TestClass.java]: File deleted.

File: src/main/java/src/chapter4/question3/Bird.java (hunk 5 above)
Comment (gabrieladutra) [reply src/main/java/src/chapter4/question3/Bird.java]: File deleted.

File: src/main/java/src/chapter4/question4/Bird.java (hunk 7 above)
Comment (gabrieladutra) [reply src/main/java/src/chapter4/question4/Bird.java]: File deleted.

File: src/test/java/chapter5/Question1Test.java (hunk 14 above)
Comment (gabrieladutra) [reply src/test/java/chapter5/Question1Test.java]: File deleted.

File: src/main/java/src/chapter5/question1/Question1.java (hunk 8 above)
Comment (gabrieladutra) [reply src/main/java/src/chapter5/question1/Question1.java]: File deleted.

File: src/main/java/src/chapter5/question4/Question4.java (hunk 11 above)
Comment (gabrieladutra) [reply src/main/java/src/chapter5/question4/Question4.java]: File deleted.

File: src/test/java/chapter4/Question2Test.java (hunk 13 above)
Comment (gabrieladutra) [reply src/test/java/chapter4/Question2Test.java]: File deleted.

File: src/test/java/chapter5/Question3Test.java (hunk 16 above)
Comment (gabrieladutra) [reply src/test/java/chapter5/Question3Test.java]: File deleted.

File: src/test/java/chapter7/Question2Test.java (hunk 18 above)
Comment (gabrieladutra) [reply src/test/java/chapter7/Question2Test.java]: File deleted.

File: src/test/java/chapter5/Question4Test.java (hunk 17 above)
Comment (gabrieladutra) [reply src/test/java/chapter5/Question4Test.java]: File deleted.

File: src/test/java/chapter5/Question2Test.java (hunk 15 above)
Comment (gabrieladutra) [reply src/test/java/chapter5/Question2Test.java]: File deleted.

File: src/main/java/src/chapter9/question2/Question2.java (hunk 12 above)
Comment (gabrieladutra) [reply src/main/java/src/chapter9/question2/Question2.java]: Fixed.

File: src/main/java/src/chapter9/Examples.java (hunk 1 above)
Comment (gabrieladutra) [reply src/main/java/src/chapter9/Examples.java:7]: Thank you!

File: src/main/java/src/chapter4/question2/TestClass.java (hunk 4 above)
Comment (gabrieladutra) [reply src/main/java/src/chapter4/question2/TestClass.java]: File deleted.

**Pull request merged**